### PR TITLE
[GOBBLIN-1882] Add brokerName in metric emitted from KafkaStreamingExtractor 

### PR DESCRIPTION
**Pull request by ZihanLi58**
Dear Gobblin maintainers,

Please accept this PR. I understand that it will not be reviewed until I have checked off all the steps below!

### JIRA
- [ ] My PR addresses the following [Gobblin JIRA](https://issues.apache.org/jira/browse/GOBBLIN/) issues and references them in the PR title. For example, "[GOBBLIN-XXX] My Gobblin PR"
    - https://issues.apache.org/jira/browse/GOBBLIN-1882

### Description
- [ ] Here are some details about my PR, including screenshots (if applicable):
Add brokerName in metric emitted from KafkaStreamingExtractor and make the kafka.brokersToSimpleNameMap to be optional so that we are backward compatible. 

### Tests
- [ ] My PR adds the following unit tests __OR__ does not need testing for this extremely good reason:
Unit tests, also test in job and make sure whether or not we specify  kafka.brokersToSimpleNameMap, it works as expected

### Commits
- [ ] My commits all reference JIRA issues in their subject lines, and I have squashed multiple commits if they address the same issue. In addition, my commits follow the guidelines from "[How to write a good git commit message](http://chris.beams.io/posts/git-commit/)":
    1. Subject is separated from body by a blank line
    2. Subject is limited to 50 characters
    3. Subject does not end with a period
    4. Subject uses the imperative mood ("add", not "adding")
    5. Body wraps at 72 characters
    6. Body explains "what" and "why", not "how"

### CODE REVIEW

File: gobblin-modules/gobblin-kafka-common/src/main/java/org/apache/gobblin/source/extractor/extract/kafka/KafkaExtractor.java [hunk 1]
```diff
@@ -354,10 +354,12 @@ public static String getKafkaBrokerSimpleName(State state) {
     String brokerUri = kafkaBrokerUriList.get(0);
     Map<String, String> brokerToSimpleName = KafkaCommonUtil.getKafkaBrokerToSimpleNameMap(state);
 
-    Preconditions.checkArgument(brokerToSimpleName.get(brokerUri) != null,
-        String.format("Unable to find simple name for the kafka cluster broker uri in the config. Please check the map "
-            + "value of %s. brokerUri=%s, configMapValue=%s", KAFKA_BROKERS_TO_SIMPLE_NAME_MAP_KEY, brokerUri, brokerToSimpleName));
-
+    if (!brokerToSimpleName.containsKey(brokerUri)) {
+      LOG.warn(String.format(
```
Comment (homatthew) [inline gobblin-modules/gobblin-kafka-common/src/main/java/org/apache/gobblin/source/extractor/extract/kafka/KafkaExtractor.java:358]: minor nit: string.format isn't necessary for logger

File: gobblin-modules/gobblin-kafka-common/src/test/java/org/apache/gobblin/source/extractor/extract/kafka/KafkaProduceRateTrackerTest.java [hunk 2]
```diff
@@ -58,6 +59,7 @@ public void setUp() {
     kafkaPartitions.add(new KafkaPartition.Builder().withTopicName("test-topic").withId(1).build());
     this.workUnitState = new WorkUnitState();
     this.workUnitState.setProp(KafkaSource.RECORD_LEVEL_SLA_MINUTES_KEY, 5L);
+    this.workUnitState.setProp(ConfigurationKeys.KAFKA_BROKERS, "testBroker");
```
Comment (homatthew) [inline gobblin-modules/gobblin-kafka-common/src/test/java/org/apache/gobblin/source/extractor/extract/kafka/KafkaProduceRateTrackerTest.java:62]: Why do we set this in the tests? Do these tests even test the eventSubmitters?

**Review by homatthew — APPROVED** [review]
some small questions. But seems good to me

File: gobblin-modules/gobblin-kafka-common/src/test/java/org/apache/gobblin/source/extractor/extract/kafka/KafkaProduceRateTrackerTest.java (hunk 2 above)
Comment (ZihanLi58) [reply gobblin-modules/gobblin-kafka-common/src/test/java/org/apache/gobblin/source/extractor/extract/kafka/KafkaProduceRateTrackerTest.java:62]: They do not, but the producerate/consumRate tracker call the util method which assume we do have broker defined.

**Pull request merged**